apps/backend/src/groups/groups.service.ts

```diff
@@ -58,19 +61,47 @@ export class GroupsService {
     const groupUser = await GroupUser.findOne({
       where: {groupId: group.id, userId: updateGroupUser.userId}
     });
+    if (groupUser) {
+      await this.setDefaultToOwner(group, groupUser.userId);
+    }
     return groupUser?.update({role: updateGroupUser.groupRole});
   }
 
   async removeUserFromGroup(group: Group, user: User): Promise<Group> {
+    await this.setDefaultToOwner(group, user.id);
+    return group.$remove('user', user);
+  }
+
+  async setDefaultToOwner(group: Group, id: string) {
```

Comment:
this function name is misleading.  default what?  why is ownership being changed on every user role modification / user removal?

i would do something closer to 'ensureGroupHasOwner' which actually mentions the relevant verb (ensuring that something is the case and making changes to have it be so if necessary), mentions the relevant noun (the group), and the relevant condition (groups must have owners).  only thing we're missing is the fact that an admin becomes the owner by default?  atm i'm not sure if this counts as an implementation detail or something worthwhile surfacing to the function name.

alternatively we could separate out the conditional from the work being done so that there would be a 'isLastOwner' function and then a 'setDefaultOwnerForGroup' function that might take an argument for which user to assign ownership to by default and then as fallback goes up the admin chain.

https://softwareengineering.stackexchange.com/questions/161753/naming-a-do-x-if-needed-method
https://softwareengineering.stackexchange.com/questions/365310/how-to-name-a-method-which-may-or-may-not-perform-an-action-depending-on-a-condi

^ interesting material to read through.  might be worthwhile having a go-back during standup or as a question posted in dev-chat on what best practices we should follow